Added documentation to README.md regarding push notifications

NunoPalma opened the pull request.

NunoPalma committed:
Merge pull request #1 from p1gp1g/unifiedpush

Unifiedpush

rfsfernandes committed:
[APY-211][COVESA SDK]: Added documentation to README.md regarding push notifications

rfsfernandes committed:
[APY-211][COVESA SDK]: Updated contributors

p1gp1g commented on README.md:
```diff
@@ -37,6 +37,142 @@ Third-party apps should link to the same single library `implementation(project(
 
 There is no AIDL layer for push notifications. Instead of an AIDL-based system service, OEMs implement their own [distributor](https://unifiedpush.org/users/distributors/) which is reacting to a specific broadcast.
 
+In order to send/receive push notifications a distributor is needed. In this case, we are using Sunup in order to be able to test. Sunup is already doing some work for us, since Mozzilla push services are set by default. We also need some kind of service that handles the Web Pushes: send, receive, register a push service and unregister push service.
```
```suggestion
In order to send/receive push notifications a distributor is needed. In this case, we are using Sunup in order to be able to test. Sunup is already doing some work for us, since Mozilla push services are set by default.

Push notifications are sent by the application server. Therefore the mobile application registers the push endpoint, and the encryption keys to this server.
```

p1gp1g commented on README.md:
````diff
@@ -37,6 +37,142 @@ Third-party apps should link to the same single library `implementation(project(
+
+Add your `PushService` implementation to your manifest:
+```xml
+<service android:name="global.covesa.sdk.client.push.PushServiceImpl"
+    android:exported="true">
+    <intent-filter>
+        <action android:name="global.covesa.sdk.PUSH_EVENT"/>
+        <action android:name="org.unifiedpush.android.connector.RAISE_TO_FOREGROUND"/>
+    </intent-filter>
+</service>
````
```suggestion
<service android:name="global.covesa.sdk.client.push.PushServiceImpl"
    android:exported="false">
    <intent-filter>
        <action android:name="global.covesa.sdk.PUSH_EVENT"/>
    </intent-filter>
</service>
```

This should not be exported, and the (dummy) service with action RAISE_TO_FOREGROUND is exposed by the lib

p1gp1g commented on README.md:
````diff
@@ -37,6 +37,142 @@ Third-party apps should link to the same single library `implementation(project(
+}
+```
+
+#### Get systems available distributors
````
Maybe we can add a comment that these methods can be used to let the user used a non-default distributor.

A UI have to be implemented to pick one of the distrib listed by getDistributors, then, once selected it is saved with saveDistributor, then it is possible to call register

rfsfernandes replied:
@p1gp1g I agree, this would help the developers. Should I develop this feature?

p1gp1g replied:
Sure, I think _Get systems available distributors_, _Save new distributor_ and _Get registered distributor_ can be under the same section, with an introduction, like _Use non-default distributor_ or something like this. It can be inspired from the [UnifiedPush documentation](https://unifiedpush.org/kdoc/connector/)

p1gp1g reviewed (commented):
Thanks, I have added a few suggestions, plus a fix for the manifest, related to #4

rfsfernandes committed:
Fixed README section about push notifications

Co-authored-by: S1m <31284753+p1gp1g@users.noreply.github.com>

rfsfernandes committed:
Fixed PushServiceImpl manifest declaration

Shouldn't be exported and RAISE_TO_FOREGROUND is no longer needed as the lib was already updated.

Co-authored-by: S1m <31284753+p1gp1g@users.noreply.github.com>

rfsfernandes committed:
[APY-211][COVESA SDK]: Updated Readme with working with non-default distributor

p1gp1g commented:
Thanks !
